provider/resource_destination.go

```diff
@@ -116,3 +119,29 @@ func resourceDestinationRead(d *schema.ResourceData, meta any) error {
 func resourceDestinationDelete(d *schema.ResourceData, meta any) error {
 	return genericResourceDelete(model.KindDestination, d, meta)
 }
+
+func resourceDestinationImportState(_ context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	bindplane := meta.(*client.BindPlane)
```

Comment:
This type assertion is safe and used all over. This is how Terraform saves arbitrary data configured during provider startup. In this case.